runtime/common/src/slots.rs

```diff
@@ -189,6 +191,30 @@ decl_module! {
 			Leases::<T>::remove(para);
 			Ok(())
 		}
+
+		/// Try to onboard a parachain that has a lease for the current lease period.
+		///
+		/// This function can be useful if there was some state issue with a para that should
+		/// have onboarded, but was unable to. As long as they have a lease period, we can
+		/// let them onboard from here.
+		///
+		/// Origin must be signed, but can be called by anyone.
+		#[weight = T::WeightInfo::trigger_onboard()]
+		fn trigger_onboard(origin, para: ParaId) {
+			let _ = ensure_signed(origin)?;
+			let leases = Leases::<T>::get(para);
+			match leases.first() {
+				// If the first element in leases is present, then it has a lease!
+				// We can try to onboard it.
+				Some(Some(_lease_info)) => {
+					T::Registrar::make_parachain(para)?
```

Comment:
should this have a `?` at the end?

Reply:
and shouldn't we be checking that it's still a plain old parathread?

Reply:
`make_parachain` itself will make sure it is a parathread and any other conditions required to allow `make_parachain` be valid. This is why we have the `?` because `make_parachain` could fail, and we will propagate up that error.